lib/spack/spack/environment/environment.py

```diff
@@ -2114,9 +2114,11 @@ def _read_lockfile_dict(self, d):
         except KeyError:
             msg = (
                 f"Spack {spack.__version__} cannot read environment lockfiles using the "
-                f"v{current_lockfile_format} format"
+                f"v{current_lockfile_format} format."
             )
-            raise RuntimeError(msg)
+            if lockfile_format_version < current_lockfile_format:
+                msg += " You may need to use a newer Spack version."
```

Comment:
I think this should be more clear and just say "you need a newer spack version"

Also does the error point to the lock file at all?  I think it should.